packages/flutter_tools/lib/src/base/common.dart

```diff
@@ -2,20 +2,43 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-/// Throw a specialized exception for expected situations
-/// where the tool should exit with a clear message to the user
-/// and no stack trace unless the --verbose option is specified.
-/// For example: network errors.
+/// Throws a specialized exception to exit with an actionable [message].
+///
+/// A [ToolExit] is interpreted by the `flutter` tool to mean "exit the tool
+/// gracefully with a human actionable error [message]", and should be used in
+/// scenarios where a human (developer) can make a decision based on the result.
+/// 
+/// A stack trace is included in the tool output when `--verbose` is specified.
+/// 
+/// For example:
+/// 
+/// - An invalid set of command-line arguments were passed.
+/// - The network appears to be unavailable.
+/// - The project needs to be modified in some way for a command to succeed.
```

Comment:
I think the examples would be easier to understand if they were directly after what they are exemplifying. Said another way, should we move ```/// A stack trace is included in the tool output when `--verbose` is specified.``` to be earlier or later in the text?

I think this should be later (maybe before the bit about not passing `null`).

Reply:
Done!